proposals/devcontainer-collections.md

```diff
@@ -0,0 +1,110 @@
+| publisher | string | The organization/user that publishes this group of definitions/features. Must match with the repository user in github.|
+| repository | string | The repository where this collection is contained.|
+| version | string | Version information of this collection list.|
+| features | array | The list of features that are contained in this collection. Same information as the metadata file for the feature.|
+| definitions | array | The list of definitions contained in this collection. Same information as the metadata file for the definition.|
+
+Creating the collections file would make some changes over the metadata of the features/definitions where they are generated.
+
+- Ids would get the collection repository added to them. (e.g. for 'https://github.com/community/features' then 'community/features' to 'myfeature1' for a feature with that id)
+- The links to the release files would be added.
+- Date.
+
+The same way creating the definitions/feature file would also be part of the release process for it so that the following metadata can be added:
+- Version.
+- Checksum.
+- Signatures.
+- Date.
+
+## Definitions/features (devcontainer-feature.json/devcontainer-definition.json)
+
+### Common Properties
+
+| Property | Type | Description |
+| :--- | :--- | :--- |
+| id | string | Id of the feature/definition. The ids should be unique in the contest of the repository where they exist. |
+| displayName | string | Name of the feature/definition |
+| shortDescription | string | Optional one sentence description |
+| description | string | Description of the feature/definition |
+| repositoryPath | string | Path to the repository and folder that contains the root of the code for the feature/definition. |
+| keywords | array | List of keywords relevant to a user that would search for this definition/feature. |
+| version | string | Versioning information for the feature/definition. |
+| files | array | List of links for the download files with checksums and optional signature information. |
+
+### Definitions
+
+| Property | Type | Description |
+| :--- | :--- | :--- |
+| type | string | singleContainer/orchestrated |
+| categories | string | The categories where it belongs |
+| architectures | string | Supported architectures |
+| baseOs | string | The operating system in which the definition is based |
+| options | array | List of user selected options at the moment the user selects them in a client application. |
+| styles | array | Variations of the generated definition that contain different information. (e.g. with example code or data.) |
+
+### Features
+
+| Property | Type | Description |
+|:--- |:--- |:--- |
+| requires | array | List of features that are required for this feature to work. |
+
+# IDs
+
+TODO
+
+# Search
+
+Search would be implemented by client applications but the definitions themselves should include enough information for this to happen. Besides the typed properties defined above keywords could be:
```

Comment:
Perhaps it is worth calling out that some keywords may be defined by clients as having an affect on UX or any "scoring" done for suggestions.

Reply:
Added that we should create a list of well-known keywords including those defined by clients.